Core/include/Acts/Seeding/SeedFilter.ipp

```diff
@@ -27,15 +27,37 @@ void SeedFilter<external_spacepoint_t>::filterSeeds_2SpFixed(
     InternalSpacePoint<external_spacepoint_t>& middleSP,
     std::vector<InternalSpacePoint<external_spacepoint_t>*>& topSpVec,
     std::vector<float>& invHelixDiameterVec,
-    std::vector<float>& impactParametersVec, float zOrigin,
-    std::back_insert_iterator<std::vector<std::pair<
-        float, std::unique_ptr<const InternalSeed<external_spacepoint_t>>>>>
+    std::vector<float>& impactParametersVec, float zOrigin, int numQualitySeeds,
+    int numSeeds,
+    std::vector<std::pair<
+        float, std::unique_ptr<const InternalSeed<external_spacepoint_t>>>>&
         outIt) const {
+  // seed confirmation
+  int nTopSeedConf = 0;
+  if (m_cfg.seedConfirmation) {
+    float rMaxSeedConfirmation =
+        std::abs(bottomSP.z()) < m_cfg.centralSeedConfirmationRange.zMaxSeedConf
+            ? m_cfg.centralSeedConfirmationRange.rMaxSeedConf
+            : m_cfg.forwardSeedConfirmationRange.rMaxSeedConf;
+    SeedConfirmationRangeConfig seedConfRange =
+        (middleSP.z() > m_cfg.centralSeedConfirmationRange.zMaxSeedConf ||
+         middleSP.z() < m_cfg.centralSeedConfirmationRange.zMinSeedConf)
+            ? m_cfg.forwardSeedConfirmationRange
+            : m_cfg.centralSeedConfirmationRange;
```

Comment:
Why do we need to check both the bottom and the middle SP here?
Can't we just check the nTopSeedConf corresponding to the central or forward confirmation range selected by the bottom sp?

Reply:
done